recipes/rmm/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+import os
+import textwrap
+from pathlib import Path
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import export_conandata_patches, get, copy, replace_in_file
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.52.0"
+
+
+class RmmConan(ConanFile):
+    name = "rmm"
+    description = "RAPIDS Memory Manager"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rapidsai/rmm"
+    topics = ("cuda", "memory-management", "memory-allocation", "rapids", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        # Based partially on https://github.com/rapidsai/rmm/tree/v23.06.00#get-rmm-dependencies
+        return {
+            "Visual Studio": "15",
+            "msvc": "191",
+            "gcc": "9.3",
+            "clang": "8",
+            "apple-clang": "14.0",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("thrust/1.17.2")
+        self.requires("spdlog/1.11.0")
+        self.requires("fmt/9.1.0")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if "backend" not in self.options["thrust"] or self.options["thrust"].backend != "cuda":
```

Comment:
```suggestion
        if self.dependencies["thrust"].options.get_safe("backend") != "cuda":
```
Accessing dependencies' options by `self.options` is no longer supported in Conan 2.x. `self.dependencies` is the new form and backward compatible with Conan 1.x